moses/moses/main/populate_atomspace.cc

```diff
@@ -10,53 +10,50 @@ namespace opencog { namespace moses {
 
 void populate(AtomSpace *as, ITable &itable) {
 
-	std::vector<multi_type_seq>::const_iterator it, end;
-	int col_size = itable.get_types().size();
-	for (int i = 0; i < col_size; i++) {
+	int n_columns = itable.get_types().size();
+	for (int i = 0; i < n_columns; i++) {
 		id::type_node col_type = itable.get_types().at(i);
-		int row_size = itable.size();
 		vertex_seq col = itable.get_column_data(i);
-		switch (col_type) {
-			case id::boolean_type: {
-				std::vector<ValuePtr> col_values = {};
-				Handle feature = createNode(SCHEMA_NODE, itable.get_labels().at(i));
-				for (int j = 0; j < row_size; j++) {
-					// for each element the ith column and the jth row
-					// change the vertex to bool
-					bool col_data = vertex_to_bool(col.at(j));
-					col_values.push_back(ValuePtr(createLink(col_data ? TRUE_LINK : FALSE_LINK)));
-				}
-				ValuePtr  vtr(new LinkValue(col_values));
-				feature->setValue(value_key, vtr);
-				as->add_atom(feature);
-				break;
-			}
-			case id::contin_type: {
-				std::vector<double> col_values_contin = {};
-				Handle feature = createNode(SCHEMA_NODE, itable.get_labels().at(i));
-				for (int j = 0; j < row_size; j++)
-					col_values_contin.push_back(
-							get_contin(col.at(j)));
+		Handle feature = createNode(SCHEMA_NODE,itable.get_labels().at(i));
+		ValuePtr vtr = vertex_seq_to_value(col, col_type);
+		feature->setValue(value_key, vtr);
+		as->add_atom(feature);
+	}
+}
 
-				ValuePtr vtr(new FloatValue(col_values_contin));
-				feature->setValue(value_key, vtr);
-				as->add_atom(feature);
-				break;
-			}
-			case id::enum_type: {
-				//TODO enum_type data to be added to Atomspace
-				break;
-			}
-			default: {
-				std::stringstream ss;
-				ss << col_type;
-				throw ComboException(TRACE_INFO,
-				                     "populate atomspace can not handle type_node %s",
-				                     ss.str().c_str());
-			}
+ValuePtr vertex_seq_to_value(const vertex_seq& col, id::type_node& col_type) {
```

Comment:
Passing `col_type` by non-const reference suggests that it will be modified, which isn't the case. Since it is an enum it could instead simply be passed by copy at no extra cost. Alternatively it could be passed by const reference, but I think best is to pass it by copy.

Reply:
Should I make it a const copy type_node since no modification is required?

Reply:
@Eman22S I'm not sure what you mean by "no modification is required". The two solutions would be
1. const reference
```c++
ValuePtr vertex_seq_to_value(const vertex_seq& col, const id::type_node& col_type);
```
2. copy
```c++
ValuePtr vertex_seq_to_value(const vertex_seq& col, id::type_node col_type);
```
I would suggest passing by copy because it's simpler and just as efficient as `type_node` amounts to an elementary type.